### PR TITLE
Allow Liquid in the recipients option

### LINKED ISSUE #1225: how to set recipient id to mail agent dynamically ?
I need to send a notification mail to different users  based on events ,Any possibility with huginn to do it .

### PR DESCRIPTION
**Pull request by cantino**
Fix #1225.

### CODE REVIEW

File: app/concerns/email_concern.rb [hunk 1]
```diff
@@ -13,7 +13,7 @@ def validate_email_options
     if options['recipients'].present?
       emails = options['recipients']
       emails = [emails] if emails.is_a?(String)
-      unless emails.all? { |email| email =~ Devise.email_regexp }
+      unless emails.all? { |email| email =~ Devise.email_regexp || email =~ /\{\{/ }
```
Comment (knu) [inline app/concerns/email_concern.rb]: Maybe just `/\{/`, to allow `{% if x %}a@x{% else %}b@y{% endif %}`.

Comment (cantino) [inline app/concerns/email_concern.rb]: That's a good point.

**Comment by jamsheer-citrus** [comment]
IS it possible with email digest agent ?

**Comment by dsander** [comment]
@jamsheer-citrus Yes that should also work with the `EmailDigestAgent`.

**Comment by jamsheer-citrus** [comment]
but its not worked to me ,Its fine for mail agent.

**Comment by dsander** [comment]
What exactly is not working? You can use liquid in the digest agent, but it will not be interpolated with the incoming events (since the digest agent can receive multiple events per digest email it would be problematic to determine which event should be used for the interpolation).

**Comment by jamsheer-thottathil** [comment]
@dsander  thank you for your reply , how to pass dynamic payload to digest agent .If i pass payload with 'message' key i am getting some extra contents in mail body as follows

example content
web_request: {"message"=>"example content"}
example content
web_request: {"message"=>"example content"} 

paylaod

{
"message": "example content"
}

my expected out put

example content

example content 

How to remove extra contents in mail body ('web_request'  like contents in above example)

**Comment by dsander** [comment]
When you are getting `web_request: {"message"=>"example content"} ` in out mails the event payload must look like this: `{ "web_request" => {"message"=>"example content"} `, you can use a `EventFormattingAgent` to transform the body to `{"message"=>"example content"}` which should then be send like to expect it.

**Comment by jamsheer-thottathil** [comment]
Thank you @dsander

**Pull request merged**